tasktiger/task.py

```diff
@@ -418,11 +418,12 @@ def cancel(self):
         """
         self._move(from_state=SCHEDULED)
 
-    def delete(self):
+    def delete(self, from_state=ERROR):
```

Comment:
The "public" `Task` API doesn't let you specify this on purpose: If it's a scheduled task, you can `cancel()` it, and if it's an errored out task you can `delete()` it, but it's unsafe to delete active tasks. It should be okay to delete queued tasks though. The tricky part is that technically a task could be in multiple states at the same time (there's nothing that prevents a unique task to be in multiple queues), and depending on how it was retrieved it may not know the state here. That's one reason on why we have two methods to get rid of a task.

Maybe we want to change the API to be simply `delete(self, from_state=None)` working as follows: `from_state = from_state or self.state; assert from_state in (QUEUED, SCHEDULED, ERROR)`, and deprecate the `cancel()` method.